### PR TITLE
Add add-karpenter-daemonset-priority-class and set-karpenter-non-cpu-limits

### PR DESCRIPTION
**Pull request by wdonne**
### add-karpenter-daemonset-priority-class

When a DaemonSet is added to a cluster every node will get a new pod. There may not be enough room for this on every node. Karpenter cannot provision extra nodes just for the DaemonSet because the new pods are not scheduled the way regular pods are. It would require parallel scheduling logic that is not proper to Kubernetes. Therefore, eviction of regular pods should happen instead. This can be achieved with the priority class system-node-critical.

### set-karpenter-non-cpu-limits

For correct node provisioning Karpenter should know exactly what the non-CPU resources are that the pods will need. Otherwise Karpenter will put as many pods on a node as possible, which may lead to memory pressure on nodes. This is especially important in consolidation mode.

## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [x] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.
- [x] I have added the artifacthub-pkg.yml file and have verified it is complete and correct.

### CODE REVIEW

**Comment by wdonne** [comment]
There is a test that fails for a strange reason. It says that the subfolder "test1" doesn't contain a YAML file with the same name as the directory plus extension. I understand that this is the case for a policy file, but my subfolders are there to host multiple tests for the same policy. With the CLI this works fine. Is there a way I can put all the tests in the same folder? As far as I can see there can only be one `kyverno-test.yaml` file and I don't see how the four tests can be hosted by it.

**Comment by mintbomb27** [comment]
@wdonne You can put all your pods in a single `resource.yaml` and then validate based on the resource name, using a single `kyverno-test.yaml`. Please find an example [here](https://github.com/kyverno/policies/tree/055cd81a7de52d66e9783bd2279854aa468d4f9a/best-practices/add-safe-to-evict).

**Comment by wdonne** [comment]
@mintbomb27 Thanks for your reply. I have updated the PR accordingly.

**Review by chipzoller — CHANGES_REQUESTED** [review]
Thanks for the PR. Just a couple minor changes.

File: karpenter/add-karpenter-daemonset-priority-class/add-karpenter-daemonset-priority-class.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: add-karpenter-daemonset-priority-class
+  annotations:
+    policies.kyverno.io/title: Add priority class for DaemonSets to help Karpenter.
+    policies.kyverno.io/subject: DaemonSet
+    policies.kyverno.io/category: Karpenter, EKS Best Practices
```
[the remark this answers is not in the document]
Comment (wdonne) [reply karpenter/add-karpenter-daemonset-priority-class/add-karpenter-daemonset-priority-class.yaml:8]: That is not yet in the Best Practices guide. It came from this discussion: https://github.com/aws/karpenter/issues/3256. I'll remove it.

File: karpenter/add-karpenter-daemonset-priority-class/artifacthub-pkg.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  pods should happen instead. This can be achieved with the priority class system-node-critical.
+install: |-
+  ```shell
+  kubectl apply -f https://raw.githubusercontent.com/kyverno/policies/main/karpenter/add-karpenter-daemonset-priority-class/add-karpenter-daemonset-priority-class.yaml
+  ```
+keywords:
+  - kyverno
+  - Karpenter
+  - EKS Best Practices
+readme: >-
```
[the remark this answers is not in the document]
Comment (wdonne) [reply karpenter/add-karpenter-daemonset-priority-class/artifacthub-pkg.yml]: I have fixed it.

File: karpenter/add-karpenter-daemonset-priority-class/daemonset.yaml
[the remark this answers is not in the document]
Comment (wdonne) [reply karpenter/add-karpenter-daemonset-priority-class/daemonset.yaml:1]: I have fixed it.

**Review by chipzoller — APPROVED** [review]
Thanks for the contributions!

**Comment by wdonne** [comment]
With pleasure!

**Pull request merged**